framework/src/modules/dynamic_rewards/module.ts

```diff
@@ -153,26 +154,36 @@ export class DynamicRewardModule extends BaseModule {
 			context.contextStore,
 			CONTEXT_STORE_KEY_BLOCK_REWARD,
 		);
-		const [blockReward, reduction] = await this._getBlockRewardDeduction(
+		let [blockReward, reduction] = await this._getBlockRewardDeduction(
 			context.getMethodContext(),
 			context.header,
 			context.assets,
 			defaultReward,
 		);
 
 		if (blockReward !== BigInt(0)) {
-			await this._tokenMethod.mint(
-				context.getMethodContext(),
+			const userSubstoreExists = await this._tokenMethod.userAccountExists(
```

Comment:
I think `mint` & `updateSharedRewards` also depend this logic ?

Reply:
@shuse2 The check should happen in `afterTransactionsExecute`. Since if it is in `beforeTransactionsExecute` there might be a case when reward module is executed earlier than token module and check for `userSubStoreExists` might assign `0` block rewards.

Reply:
@vjaiman this will be handled in https://github.com/LiskHQ/lisk-sdk/issues/8194